Don't resolve Callable NamedTuple fields to their return type

FuegoFro opened the pull request:
`NamedTuple` fields are always marked an properties. In certain situations, this causes their type to be determined incorrectly when they hold `Callable`s, resolving to the `Callable`'s return type rather than the `Callable` itself. This prevents that and fixes https://github.com/python/mypy/issues/6575.

msullivan commented:
I am going to look at the bug you filed and think a bit about the whether this is the best approach (or whether, as you suggested, we should change something on the namedtuple side setting properties), but separate from that: could you please add a test.

msullivan commented on mypy/subtypes.py:
```diff
@@ -622,7 +622,7 @@ def find_node_type(node: Union[Var, FuncBase], itype: Instance, subtype: Type) -
         assert isinstance(typ, FunctionLike)
         signature = bind_self(typ, subtype)
-        if node.is_property:
+        if node.is_property and not node.info.is_named_tuple:
```
Hm, it looks like this will still do `bind_self` on the type, which I think might cause trouble, since it will modify the type of the field in an unexpected way. Make sure to test with multiple arguments.

msullivan replied:
Looking at https://github.com/python/mypy/blob/master/mypy/checkmember.py#L495, it seems that only  `is_initialized_in_class` vars get the property/self binding treatment, so probably the right approach is to match that.

FuegoFro commented:
Thanks for the response! Yeah, I'll take a stab at adding some tests for various scenarios, including where the `Callable` takes multiple arguments 🙂

FuegoFro commented:
Thanks for the pointers here! I changed the 'self' binding check to match the code/condition you linked to 🙂 I also added some integration tests. I haven't really worked with this codebase much before, so I wasn't sure if there's a way I should try to make these work in the [`testsubtypes.py`](https://github.com/python/mypy/blob/master/mypy/test/testsubtypes.py) file. Let me know if there's a better way to add tests, or any other types of tests you'd like to see!

I did notice that a `TestPEP561.test_typedpkg_editable` test was failing locally, but it failed for me even on master and it seems unrelated to this change. However, if that does seem relevant and you have some pointers I can dig into it more 🙂

FuegoFro commented:
Thank you for you help and quick responses/turnaround! Don't think I was in a `venv`, but it sounds like this wasn't entirely unexpected, so I'll hold off on looking into my local setup until/unless I have more PRs to send 😉
